src/Bicep.Core/Parsing/BaseParser.cs

```diff
@@ -561,6 +571,8 @@ private IEnumerable<SyntaxBase> HandleFunctionElements(TokenType closingTokenTyp
 
             itemsOrTokens.AddRange(NewLines());
 
+            var isClosed = FunctionElementListIsClosed(openingTokenType, closingTokenType);
```

Comment:
Is it possible to avoid doing this unneccesarily?

Reply:
I couldn't come up with a way to avoid this, but it's normally a cheap operation because it only scans forward to the closing paren/chevron.